### PR TITLE
when cloning an order, payment rule should be set as in the original …

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/12506

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java [hunk 1]
```diff
@@ -230,17 +230,20 @@ public void setDeliveryViaRule(final I_C_Order order)
...
 	public void setPaymentRule(final I_C_Order order)
 	{
-		final I_C_BPartner bpartner = order.getC_BPartner();
-		final PaymentRule paymentRule;
-		if (bpartner != null && bpartner.getPaymentRule() != null)
-		{
-			paymentRule = PaymentRule.ofCode(bpartner.getPaymentRule());
-		}
-		else
+		if (!InterfaceWrapperHelper.isCopying(order))
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java:233]: shouldn't we then have this if also in `backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java` ?

Comment (metas-ts) [reply backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java:233]: shouldn't we also have `&& !InterfaceWrapperHelper.isNew(invoice)` here?

Comment (riadmen) [reply backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java:233]: If I understood correctly from Daniela::

- A new order is supposed to inherit the payment rule from the BPartener:
- A new invoice is supposed to inherit the payment rule from the related order.

so i think we shouldn't add it to `de.metas.order.model.interceptor.C_Order`

File: backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java [hunk 2]
```diff
@@ -177,17 +177,20 @@ public void removeMaterialLinesNotCorrespondingToPriceList(final I_C_Invoice inv
 	@CalloutMethod(columnNames = I_C_Invoice.COLUMNNAME_C_BPartner_ID)
 	public void setPaymentRule(final I_C_Invoice invoice)
 	{
-		final I_C_BPartner bpartner = bpartnerDAO.getById(invoice.getC_BPartner_ID());
-		final PaymentRule paymentRule;
-		if (bpartner != null && bpartner.getPaymentRule() != null)
+		if (!InterfaceWrapperHelper.isCopying(invoice) && !InterfaceWrapperHelper.isNew(invoice))
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java:180]: please add javadoc to explain why this method only acts if (!InterfaceWrapperHelper.isCopying(invoice) && !InterfaceWrapperHelper.isNew(invoice))`

❗ btw, what about the case that a new invoice without an order is created in the webui?

File: backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java (hunk 1 above)
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_Order.java:233]: please add javadoc to explain why this method only acts `if (!InterfaceWrapperHelper.isCopying(order))`

**Review by metas-ts — CHANGES_REQUESTED** [review]

**Review by metas-ts — APPROVED** [review]

**Pull request merged**